File: 04_07_LCMandGCD.py
class Solution:
    def lcmAndGcd(self, a : int, b : int) -> List[int]:
        product = a*b
        while a>0 and b>0:
            if a > b:
                a = a % b
            else:
                b = b % a
        if a == 0:
            gcd = b
        else:
            gcd = a
        return [int(product/gcd), gcd]

File: Module_1/04_07_LCMandGCD.cpp
class Solution {
  public:
    vector<int> lcmAndGcd(int a, int b) {
        
        int gcd;
        vector<int> result;
        int product = a * b;
        
        while(a>0 && b>0) {
            if(a > b) {
                a = a % b;
            } else {
                b = b % a;
            }
        }
        
        if(a == 0) {
            gcd = b;
        } else {
            gcd = a;
        }
        
        result.push_back(product/gcd);
        result.push_back(gcd);
        
        return result;
    }
};
